src/main/java/core/basesyntax/dao/impl/UserDaoImpl.java

```diff
@@ -12,21 +15,63 @@ public UserDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public User create(User entity) {
-        return null;
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(entity);
+            transaction.commit();
+            return entity;
+        } catch (RuntimeException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't add user entity " + entity, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public User get(Long id) {
-        return null;
+        try (Session session = factory.openSession()) {
+            return session.get(User.class, id);
+        } catch (RuntimeException e) {
+            throw new RuntimeException("Can't get user with id = " + id, e);
+        }
     }
 
     @Override
     public List<User> getAll() {
-        return null;
+        try (Session session = factory.openSession()) {
+            Query<User> getAll = session.createQuery("from User", User.class);
+            return getAll.getResultList();
+        } catch (RuntimeException e) {
+            throw new RuntimeException("Can't get all users", e);
+        }
     }
 
     @Override
     public void remove(User entity) {
-
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.delete(entity);
```

Comment:
it only for your understanding: read about the difference between remove() and delete() methods in the hibernate